### PR TITLE
fix: description dereferenced

### PR DESCRIPTION
**Pull request by iosypov**
## What/Why/How?

Fix: #750. Make fix in object assign. Rewrite all info from parent, additional from $ref.

## Screenshots (optional)

image

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/bundle.ts [hunk 1]
```diff
@@ -304,7 +304,8 @@ function makeBundleVisitor(
     } else {
       // @ts-ignore
       delete ref.$ref;
-      Object.assign(ref, resolved.node);
+      const obj = Object.assign({}, resolved.node, ref);
+      Object.assign(ref, obj);
```
Comment (RomanHotsiy) [inline packages/core/src/bundle.ts:308]:
```suggestion
      Object.assign(ref, resolved.node, ref); // assign ref itself again so ref fields take precedence
```

**Review by RomanHotsiy — APPROVED** [review]
We need to add tests for this case too.

**Review by tatomyr — APPROVED** [review]

File: packages/core/src/bundle.ts (hunk 1 above)
Comment (iosypov) [reply packages/core/src/bundle.ts]: It's wrong, because we should get first val info from resolved.node.

Comment (RomanHotsiy) [reply packages/core/src/bundle.ts]: It's the same. I don't see how it's different. Can you show me some counter-example?

**Review by RomanHotsiy — APPROVED** [review]

**Pull request merged**